proxy/tests/support/client.rs

```diff
@@ -102,6 +102,9 @@ fn run(addr: SocketAddr, version: Run) -> Sender {
                     .build(&reactor);
                 Box::new(rx.for_each(move |(req, cb)| {
                     let mut req = hyper::Request::from(req.map(|()| hyper::Body::empty()));
+                    if !req.headers().has::<hyper::header::ContentLength>() {
+                        assert!(req.body_mut().take().unwrap().is_empty());
```

Comment:
I think this should be:

```rust
assert_eq!(!req.body_mut().take().unwrap().is_empty(), req.headers().has::<hyper::header::ContentLength>());
```

Shouldn't we be considering the presence of Transfer-Encoding as an alternative to Content-Length here too?